lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -202,6 +202,21 @@ forwardKnowledgeTransferFunction(Operation *op,
     // the same, then the result is of that same element type.
     knowledge.elementType = joinElementTypes(lhs.elementType, rhs.elementType);
     return {knowledge};
+  } else if (isa<aten::LinearOp>(op)) {
+    // The output shape is the input shape with the last dimension changed
+    // to the weight's output dimension.
+    auto knowledge = operandKnowledge[0];
+    if (knowledge.hasRank && knowledge.sizes.size() > 0)
+      knowledge.sizes[knowledge.sizes.size() - 1] = kUnknownSize;
+    // TODO: Handle case of bias being None gracefully. Requires a lattice
```

Comment:
Wdyt of adding an issue category for starter issues? It seems like this would be a good one to note there.

Reply:
Filed https://github.com/llvm/mlir-npcomp/issues/211

I'm hoping to roll this into the new "aten-to-linalg + shape inference + aten/prim td generation + named op generation" component handles.